### PR TITLE
Add test that demonstrates usage of the maeparser

### PR DESCRIPTION
**Pull request by d-b-w**
Shows the kind of data available from an Mae file, and the
basics of accessing that data.

### CODE REVIEW

File: test/UsageDemo.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        {
+            const auto bond_data = b->getIndexedBlock("m_bond");
+            // All bonds are gauranteed to have these three field names:
+            auto from_atoms = bond_data->getIntProperty("i_m_from");
+            auto to_atoms = bond_data->getIntProperty("i_m_to");
+            auto orders = bond_data->getIntProperty("i_m_order");
+            const auto size = from_atoms->size();
+
+            for (size_t i=0; i<size; ++i) {
+                // Maestro atoms are 1 indexed!
```
Comment (mbeachy) [inline test/UsageDemo.cpp]: I think this would be more clear if you said "Atom indices in the bond data are 1-based!"

File: test/UsageDemo.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            BOOST_REQUIRE_EQUAL(size, ys->size());
+            BOOST_REQUIRE_EQUAL(size, zs->size());
+
+            // atomic numbers, and x, y, and z coordinates
+            for (size_t i=0; i<size; ++i) {
+                st->atomic_numbers.push_back(atomic_numbers->at(i));
+                st->coordinates.push_back({{xs->at(i), ys->at(i), zs->at(i)}});
+
+                // Other properties could fail, because not all properties have
+                // values for all atoms.
```
Comment (mbeachy) [inline test/UsageDemo.cpp]: Maybe "Use isDefined(index) to determine when this is the case."

File: test/UsageDemo.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
+///
+// Demonstrate reading a schrodinger/Maestro formatted file and gleaning the
+// bonding information, coordinates, and atomic number.
+//
+// Note that Maestro format uses 1-based indices.
```
Comment (mbeachy) [inline test/UsageDemo.cpp]: ...when referring to indexed-data in blocks.

File: test/UsageDemo.cpp [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            // All bonds are gauranteed to have these three field names:
+            auto from_atoms = bond_data->getIntProperty("i_m_from");
+            auto to_atoms = bond_data->getIntProperty("i_m_to");
+            auto orders = bond_data->getIntProperty("i_m_order");
+            const auto size = from_atoms->size();
+
+            for (size_t i=0; i<size; ++i) {
+                // Maestro atoms are 1 indexed!
+                const auto from_atom = from_atoms->at(i) - 1;
+                const auto to_atom = to_atoms->at(i) - 1;
```
Comment (mbeachy) [inline test/UsageDemo.cpp]: Hmmm. Should we add a getIndexProperty method that does this automatically for its values?

**Comment by d-b-w** [comment]
Mike-
I've updated the diff associated with this PR. It looks like, in the future,  the hip thing to do in github would have been to add a second commit and let you squash them when merging.

**Pull request merged**